rest-api/test/unit_test/dao_test/participant_summary_dao_test.py

```diff
@@ -292,6 +294,36 @@ def add_sample(test_code, sample_id):
     self.assertEquals(summary.numBaselineSamplesArrived, 1)
     self.assertNotEqual(init_last_modified, summary.lastModified)
 
+  def test_only_update_dna_sample(self):
+    dna_tests = ["1ED10", "1SAL2"]
+
+    config.override_setting(config.DNA_SAMPLE_TEST_CODES, dna_tests)
+    self.dao.update_from_biobank_stored_samples()  # safe noop
+
+    p_dna_samples = self._insert(Participant(participantId=1, biobankId=11))
+
+    self.assertEquals(self.dao.get(p_dna_samples.participantId).samplesToIsolateDNA, None)
+    self.assertEquals(
+      self.dao.get(p_dna_samples.participantId).enrollmentStatusCoreStoredSampleTime, None)
+
+    sample_dao = BiobankStoredSampleDao()
+
+    def add_sample(participant, test_code, sample_id, confirmed_time):
+      sample_dao.insert(BiobankStoredSample(
+          biobankStoredSampleId=sample_id, biobankId=participant.biobankId,
+          biobankOrderIdentifier='KIT', test=test_code, confirmed=confirmed_time))
+
+    confirmed_time_0 = datetime.datetime(2018, 3, 1)
+    add_sample(p_dna_samples, dna_tests[0], '11111', confirmed_time_0)
+
+    self.dao.update_from_biobank_stored_samples()
+
+    self.assertEquals(self.dao.get(p_dna_samples.participantId).samplesToIsolateDNA,
+                      SampleStatus.RECEIVED)
+    # only update dna sample will not update enrollmentStatusCoreStoredSampleTime
+    self.assertEquals(
```

Comment:
We should probably have tests somewhere that check that enrollment status gets updated appropriately when canceling PMs or biobank orders... maybe updating:

https://github.com/all-of-us/raw-data-repository/blob/master/rest-api/test/unit_test/api_test/biobank_order_api_test.py#L58

https://github.com/all-of-us/raw-data-repository/blob/master/rest-api/test/unit_test/api_test/physical_measurements_api_test.py#L266

Reply:
These two test cases are already in participant_summary_api_test.py